src/app/state/rewardSlice.ts

```diff
@@ -0,0 +1,238 @@
+    return name?.entries[0].value.entries[0].value.value;
+  } catch (error) {
+    console.log(error);
+    return undefined;
+  }
+
+  return undefined;
+});
+
+export const rewardSlice = createSlice({
+  name: "reward",
+  initialState,
+
+  // synchronous reducers
+  reducers: {
+    claimRewards: (state) => {
+      const rdt = getRdt();
+      if (!rdt) return;
+
+      const recieptIds = state.recieptIds;
+      if (recieptIds === null) return;
+      if (recieptIds.length === 0) return;
+
+      const walletData = rdt.walletApi.getWalletData();
+      const accountAddress = walletData.accounts[0].address;
+      const resourceAddress =
+        process.env.NEXT_PUBLIC_RESOURCE_ADDRESS_DEXTERXRD;
+      const claimComponentAddress = process.env.NEXT_PUBLIC_CLAIM_COMPONENT;
+      const claimNFTAddress = process.env.NEXT_PUBLIC_CLAIM_NFT_ADDRESS;
+      const resourcePrefix = process.env.NEXT_PUBLIC_RESOURCE_PREFIX;
+      if (!resourcePrefix) return;
+      const nonfungibleLocalId = accountAddress.replace(
+        new RegExp(resourcePrefix, "g"),
+        ""
+      );
+
+      const nftArray = recieptIds
+        .map((id) => `NonFungibleLocalId("${id}")`)
+        .join(",");
+
+      const claimManifest = `
+        CALL_METHOD 
+          Address("${accountAddress}") 
+          "create_proof_of_non_fungibles" 
+          Address("${claimNFTAddress}") 
+          Array<NonFungibleLocalId>(NonFungibleLocalId("<${nonfungibleLocalId}>")); 
+        POP_FROM_AUTH_ZONE 
+          Proof("account_proof_1");
+          CALL_METHOD Address("${accountAddress}") 
+          "create_proof_of_non_fungibles" 
+          Address("${resourceAddress}") 
+          Array<NonFungibleLocalId>(${nftArray}); 
+        CREATE_PROOF_FROM_AUTH_ZONE_OF_ALL 
+          Address("${resourceAddress}")  
+          Proof("proof_1");
+        CALL_METHOD 
+          Address("${claimComponentAddress}") 
+          "claim_rewards" 
+          Array<Proof>(Proof("account_proof_1")) 
+          Array<Proof>(Proof("proof_1"));
+        CALL_METHOD 
+          Address("${accountAddress}") 
+          "deposit_batch" 
+          Expression("ENTIRE_WORKTOP");
+        `;
+
+      rdt.walletApi.sendTransaction({
+        transactionManifest: claimManifest,
+      });
+    },
+    getEarnedRewards: (state) => {
+      const rdt = getRdt();
+      if (!rdt) return;
+
+      const recieptIds = state.recieptIds;
+      if (recieptIds === null) return;
+      if (recieptIds.length === 0) return;
+
+      const walletData = rdt.walletApi.getWalletData();
+      const accountAddress = walletData.accounts[0].address;
+      const resourceAddress =
+        process.env.NEXT_PUBLIC_RESOURCE_ADDRESS_DEXTERXRD;
+      const claimComponentAddress = process.env.NEXT_PUBLIC_CLAIM_COMPONENT;
+      const claimNFTAddress = process.env.NEXT_PUBLIC_CLAIM_NFT_ADDRESS;
+      const resourcePrefix = process.env.NEXT_PUBLIC_RESOURCE_PREFIX;
```

Comment:
🤔  should we have fallback testnet values for these directly in code (here and in other places)? like it is with the other env vars

another idea (but probably for a separate PR) - having only one env var with the value `testnet` or `mainnet` with testnet as default, and depending on this set all the other variables (either as env vars or TS variables), somewhere in build time... not sure where exactly yet, maybe in next settings, this would:
- help organize the thing - there will never be a missing variable and new contributors can quickly launch the project
- reduce code 
- no need to duplicate this info in CONTRIBUTING.md - all the values will be accessible